### PR TITLE
Improving Doctor Connect UI

### LINKED ISSUE #6940: Multiple scrolls- unnecessary: doctor connect
**Describe the bug**
When there are over 5 staff of a particular type appearing on the doctor connect panel, the section (Doctor/staff/teleicu Hub) has a separate scroller while the whole Doctor connect panel has a second scroller. This seems confusing and unnecessery.

**To Reproduce**

1. Go to 'any patient with consultation filed'
2. Click on '.Doctor connect.'
3. if there are over 5 listing in any of the section, you will see a section scroller in addition to the doctor connect panel scroller.

Screenshot 2023-12-29 at 5 44 01 PM

This was done maybe to retain all the section headings in view on the panel.

Can we think of a better visualisation? Maybe collapsable sections?

### PR DESCRIPTION
**Pull request by thedevildude**
## Proposed Changes

- Fixes #6940
- Redesigned the UI based on mockup
- Added User Filter logic and Switches

| Before | After |
| ------- | ------ |
| image |image|

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

File: src/Components/Facility/DoctorVideoSlideover.tsx [hunk 1]
```diff
@@ -1,48 +1,75 @@
...
-      fetchUsers();
+    if (users?.results && !loading) {
+      setFilteredDoctors(
+        filterDoctors(users.results).sort(
+          (a: UserAssignedModel, b: UserAssignedModel) => {
+            const aIsHomeUser = isHomeUser(a, facilityId);
+            const bIsHomeUser = isHomeUser(b, facilityId);
+            return aIsHomeUser === bIsHomeUser
+              ? 0
+              : isHomeUser(a, facilityId)
```
Comment (rithviknishad) [inline src/Components/Facility/DoctorVideoSlideover.tsx]:
```suggestion
              : aIsHomeUser
```

**Review by Ashesh3 — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@thedevildude 

image

- [ ] In the new UI positioning of whatsapp and phone icons seem so weird, i would prefer the current position itself ( Personal Opinion not a suggestion )
- [ ] Think of a better UI for the card, and get it approved in the eod call before reworking this PR again
- [ ] fix the failing cypress test

**Comment by thedevildude** [comment]
@nihal467 will do

File: src/Redux/api.tsx [hunk 2]
```diff
@@ -354,7 +355,7 @@ const routes = {
 
   getFacilityUsers: {
     path: "/api/v1/facility/{facility_id}/get_users/",
-    TRes: Type<PaginatedResponse<UserModel>>(),
+    TRes: Type<PaginatedResponse<UserAssignedModel>>(),
```
Comment (rithviknishad) [inline src/Redux/api.tsx:358]: Why was this changed?

`UserAssignedModel` contains less information than what the response gives and is not the right type. Revert it to `UserModel`.
```suggestion
    TRes: Type<PaginatedResponse<UserModel>>(),
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Redux/api.tsx (hunk 2 above)
Comment (thedevildude) [reply src/Redux/api.tsx:358]: @rithviknishad UserModel doesn't have the skills attribute. When we are fetching the facility users here, we are also receiving the user's skills.
image

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by Ashesh3 — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]
